### PR TITLE
New Maintainer Spring clean: Dependency and Github workflow updates

### PR DESCRIPTION
**Pull request by henricook**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,17 +1,18 @@
 ThisBuild / tlBaseVersion := "0.23"
-ThisBuild / tlMimaPreviousVersions ++= (0 to 11).map(y => s"0.23.$y").toSet
+ThisBuild / tlMimaPreviousVersions ++= (0 to 23).map(y => s"0.23.$y").toSet
```
Comment (armanbilge) [inline build.sbt:2]: This should stay fixed at 11. From 0.23.12+ we no longer published http4s-play-json from the http4s/http4s repo.

Comment (henricook) [reply build.sbt]: Ah yes I see, correct I believe? Thanks

**Comment by henricook** [comment]
@armanbilge Green build - WDYT? I'd look to release 0.23.23 after this gets merged. Have I missed anything that's required before making a new release in GH?

**Comment by armanbilge** [comment]
> I'd look to release 0.23.23

I think the next version is 0.23.12, since the last release was 0.23.11.

**Comment by henricook** [comment]
> > I'd look to release 0.23.23
> 
> I think the next version is 0.23.12, since the last release was 0.23.11.

Ah, I thought I should keep releasing in sync with http4s. If it's fine to be out of sync with http4s then all good, can do 0.23.12.

File: build.sbt [hunk 2]
```diff
@@ -1,17 +1,21 @@
 ThisBuild / tlBaseVersion := "0.23"
 ThisBuild / tlMimaPreviousVersions ++= (0 to 11).map(y => s"0.23.$y").toSet
 ThisBuild / developers := List(
-  tlGitHubDev("rossabaker", "Ross A. Baker")
+  tlGitHubDev("rossabaker", "Ross A. Baker"),
+  tlGitHubDev("henricook", "Henri Cook"),
 )
 
-val Scala213 = "2.13.10"
-ThisBuild / crossScalaVersions := Seq("2.12.17", Scala213)
```
Comment (armanbilge) [inline build.sbt:8]: Scala 2.12 is still supported in play-json 2.10.1, isn't it a bit early to drop it?

Meanwhile, we can add Scala 3 now?

**Comment by armanbilge** [comment]
> If it's fine to be out of sync with http4s then all good, can do 0.23.12.

Yes, there's no reason to stay in sync, and in the long term it will not be sustainable. What would happen if you need to release a bug fix, how can you do it without http4s releasing first?

File: build.sbt (hunk 2 above)
Comment (henricook) [reply build.sbt:8]: I think that's a mistake on their part, as they looked to be preparing things to be matching Play 2.9's release which drops 2.12. BUT - you're right, until they actually do - we can keep supporting. Will amend and add Scala 3, good call.

Comment (henricook) [reply build.sbt:8]: Amended

**Review by armanbilge — APPROVED** [review]
nice one!

**Pull request merged**